lib/config/configuration.go

```diff
@@ -891,6 +891,16 @@ func applyProxyConfig(fc *FileConfig, cfg *servicecfg.Config) error {
 		cfg.Proxy.UI = webclient.UIConfig(*fc.Proxy.UI)
 	}
 
+	if fc.Proxy.Assist != nil && fc.Proxy.Assist.OpenAI != nil {
```

Comment:
Should we still add an `assist.enabled` flag, and not just check whether a static key is present?

I understand that this makes it harder to spot the error (have to wait until assistant is invoked to detect that the feature is enabled, but the key is missing), but when we implement user-supplied keys via resources (OSS or Cloud, or both), there might be no static key in YAML, but the on/off flag may still be desirable.

Reply:
Another question: if this static token field is hard-required, what value do we put in here in Cloud?

I guess there's no harm to reference the same resource file both here, and in the resource, just a bit weird.

Reply:
I don't think we need the `enabled` switch. Assist can be only enabled when the API key is provided, so that would be just an additional boilerplate that we need to keep.

>Another question: if this static token field is hard-required, what value do we put in here in Cloud?

I wanted to make it "hard-required" only on OSS. Cloud IMO should only be dependent on the plugin resource. I was thinking about checking the plugin first, and if it doesn't exist, then check the config file. We could also probably guard it with `!isCloud()` or something similar. 

My assumption is that we will remove that API key from our config soon and start using plugins on OSS and Cloud that is why I want to keep the config very minimal.

Reply:
> I was thinking about checking the plugin first, and if it doesn't exist, then check the config file.

Hmm yeah, when we remove the static config field later on, the question in Proxy "is assist enabled" will get more complicated, and we might have to check for resource existence even in `/ping`.

Anyway, I think it's fine for now. Cloud will have to also specify the key in static config.